prism/src/main/java/org/apache/falcon/resource/AbstractEntityManager.java

```diff
@@ -365,6 +374,81 @@ protected APIResult update(Entity newEntity, String type, String entityName, Boo
     }
 
     /**
+     * check if the data source entity dependent feeds are upgraded or not by checking against the data source entity
+     * version and upgrade feeds accordingly.
+     *
+     * @param datasourceName Name of the data source entity
+     * @param skipDryRun Skip dry run during update if set to true
+     * @return APIResult
+     *
+     */
+
+    public APIResult updateDatasourceDependents(String datasourceName, Boolean skipDryRun) {
+        try {
+            Datasource datasource = EntityUtil.getEntity(EntityType.DATASOURCE, datasourceName);
+            StringBuilder result = new StringBuilder(String.format("Updating feed entities "
+                    + "dependent on datasource : %s ", datasource.getName()));
+
+            // get data source dependent entities and check the version referenced is same
+            Pair<String, EntityType>[] dependentEntities = EntityIntegrityChecker.referencedBy(datasource);
+            if (dependentEntities == null) {
+                return new APIResult(APIResult.Status.SUCCEEDED, String.format("Datasource %s has "
+                        + "no dependent entities", datasourceName));
+            }
+            for (Pair<String, EntityType> depEntity : dependentEntities) {
+                Entity entity = EntityUtil.getEntity(depEntity.second, depEntity.first);
+                if (entity.getEntityType() == EntityType.FEED) {
+                    Feed newFeed = (Feed) entity.copy();
+                    for (org.apache.falcon.entity.v0.feed.Cluster feedCluster
+                            : newFeed.getClusters().getClusters()) {
+                        if (feedCluster.getType() == ClusterType.SOURCE) {
+                            boolean updatedFeed = isUpdateFeedDatasourceVersion(feedCluster, datasource, newFeed);
+                            if (updatedFeed) {
+                                // rewrite the dependent feed and update it on the store
+                                result.append(getWorkflowEngine(entity).update(entity, newFeed,
+                                        feedCluster.getName(), skipDryRun));
+                                updateEntityInConfigStore(entity, newFeed);
+                            }
+                        }
+                    }
+                }
+            }
```

Comment:
Why check if type is FEED? isnt it always feed? Should we throw an error if it is not FEED?